.github/workflows/integration.yml

```diff
@@ -20,13 +20,23 @@ jobs:
       - uses: actions/setup-go@v4
         with:
           go-version: ${{ env.GO_VERSION }}
+      - run: sudo chown -R $(id -u):$(id -g) /root
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/go/pkg/mod
```

Comment:
I think a somewhat similar change has been removed recently in [this](https://github.com/cri-o/cri-o/commit/aa86e94f04c806b26abfafb323a94ab470de66d1) commit.

Reply:
actions/setup-go@v4 caches a few directories -- `$(go env GOMODCACHE)` and `$(go env GOCACHE)`, to be precise, those are usually `~/go/pkg/mod` and `~/.cache/go-build`.

I think we should build all sources and tests as non-root (and I don't see any cases where we run `go build` or `go test` as root, maybe I'm mistaken?), so the above caching should be sufficient.